### PR TITLE
Update scatter.glsl to fix issue with textures

### LINKED ISSUE #36: Rendering Error with RDNA2
I noticed a weird rendering error in scene 2:
Metallic
Swapping the material of the affected sphere out to  `Material::Lambertian(vec3(1.0f),1)` resolves the issue:
grafik
Switching the center Spere to `Material::Metallic(vec3(1.0f), 0.0f)` causes the issue to also affect this sphere:
grafik

This is on an RX 6800 on the 21.2.2 driver.

### PR DESCRIPTION
**Pull request by PMunkes**
This proposed change fixes #36 without any visual impact.

### CODE REVIEW

**Comment by GPSnoopy** [comment]
Hi @PMunkes , thanks for looking into this.

I'll come back to it in a few weeks, family matters to attend to first. But I wanted you to know are not being ignored :-)

**Comment by PMunkes** [comment]
Hi @GPSnoopy, take your time, there is no urgency here.

I really enjoy experimenting with this project, so don't be surprised if there's another Pull Request when you return.

**Comment by PMunkes** [comment]
Updated the pull request since I found the root cause.
The problem was that the texture indices were not specified to be nonuniform.
https://github.com/KhronosGroup/GLSL/blob/master/extensions/ext/GL_EXT_nonuniform_qualifier.txt

File: assets/shaders/Scatter.glsl [hunk 1]
```diff
@@ -13,7 +13,7 @@ float Schlick(const float cosine, const float refractionIndex)
 RayPayload ScatterLambertian(const Material m, const vec3 direction, const vec3 normal, const vec2 texCoord, const float t, inout uint seed)
 {
 	const bool isScattered = dot(direction, normal) < 0;
-	const vec4 texColor = m.DiffuseTextureId >= 0 ? texture(TextureSamplers[m.DiffuseTextureId], texCoord) : vec4(1);
+	const vec4 texColor = m.DiffuseTextureId >= 0 ? texture(TextureSamplers[nonuniformEXT(m.DiffuseTextureId)], texCoord) : vec4(1);
```
Comment (GPSnoopy) [inline assets/shaders/Scatter.glsl:16]: Shouldn't the top of the file ideally have a `#extension GL_EXT_nonuniform_qualifier: require` ?

Comment (PMunkes) [reply assets/shaders/Scatter.glsl:20]: Probably, I will include the extensions.

**Comment by PMunkes** [comment]
Found a very interesting link, in case anyone finds this later and wants more information:
https://gpuopen.com/learn/porting-detroit-2/

**Pull request merged**